===== Patch instructions generated by SAVER =====
[+] { Insert: if (true) free(*(init:h)) at 16 (line 71, column 14),
===== Actual patch in diff format =====
--- a/table1/benchmarks/x264/filters/video/cache.c
+++ b/table1/benchmarks/x264/filters/video/cache.c
@@ -69,6 +69,7 @@ static int init( hnd_t *handle, cli_vid_filter_t *filter, video_info_t *info, x2
     {
         h->cache[i] = malloc( sizeof(cli_pic_t) );
         if( !h->cache[i] || x264_cli_pic_alloc( h->cache[i], info->csp, info->width, info->height ) )
+            free(h);
             return -1;
     }
     h->cache[h->max_size] = NULL; /* require null terminator for list methods */
